packages/replay/src/auth.ts

```diff
@@ -263,3 +264,128 @@ export async function maybeAuthenticateUser(options: Options = {}) {
     return false;
   }
 }
+
+function parseId(id: string) {
+  return Buffer.from(id, "base64").toString("utf-8").split(":")[1];
+}
+
+async function getAuthInfo(key: string) {
+  const resp = await query(
+    "AuthInfo",
+    `
+        query AuthInfo {
+          viewer {
+            user {
+              id
+            }
+          }
+          auth {
+            workspaces {
+              edges {
+                node {
+                  id
+                }
+              }
+            }
+          }
+        }
+      `,
+    undefined,
+    key
+  );
+
+  if (resp.errors) {
+    throw {
+      id: "graphql-error",
+      message: resp.errors
+        .map((e: any) => e.message)
+        .filter(Boolean)
+        .join(", "),
+    };
+  }
+
+  const response = resp.data as {
+    viewer: {
+      user: {
+        id: string | null;
+      } | null;
+    };
+    auth: {
+      workspaces: {
+        edges: {
+          node: {
+            id: string;
+          };
+        }[];
+      };
+    };
+  };
+
+  const encodedUserId = response?.viewer?.user?.id;
+  const encodedWorkspaceId = response?.auth?.workspaces?.edges?.[0]?.node?.id;
```

Comment:
This is probably too defensive and we should check more strictly earlier so we can report on unexpected conditions.